pkg/agent/agentversionsjob.go

```diff
@@ -0,0 +1,201 @@
+	"EnterpriseEdgeGatewayTraceabilityAgent": "v7-traceability-agent",
+}
+
+type htmlAnchors struct {
+	VersionList []string `xml:"body>pre>a"`
+}
+
+type version struct {
+	major, minor, patch int
+	val                 string
+}
+
+// agentVersionCheckJob - polls for agent versions
+type agentVersionCheckJob struct {
+	jobs.Job
+	allVersions   []string
+	latestVersion string
+	buildVersion  string
+	dataPlaneType string
+	urlName       string
+}
+
+// Ready -
+func (avj *agentVersionCheckJob) Ready() bool {
+	avj.setURLName()
+	return true
+}
+
+// Status -
+func (avj *agentVersionCheckJob) Status() error {
+	return nil
+}
+
+// Execute - run agent version check job one time
+func (avj *agentVersionCheckJob) Execute() error {
+	if avj.urlName == "AgentSDK" || avj.urlName == "" {
+		err := errors.ErrStartingVersionChecker.FormatError("empty or generic data plane type name")
+		log.Error(err)
+		return err
+	}
+	err := avj.getBuildVersion()
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	err = avj.getJFrogVersions(avj.urlName)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	// compare build to latest version
+	if isVersionStringOlder(avj.buildVersion, avj.latestVersion) {
+		log.Infof("Running older version of %s. Please consider upgrading from version %s to version %s", avj.dataPlaneType, avj.buildVersion, avj.latestVersion)
+	}
+	return nil
+}
+
+func (avj *agentVersionCheckJob) getBuildVersion() error {
+	avj.buildVersion = agent.cfg.GetBuildVersion()
+	//remove -SHA from build version
+	noSHA := strings.Split(avj.buildVersion, "-")
+	avj.buildVersion = noSHA[0]
+
+	//regex check for semantic versioning
+	semVerRegexp := regexp.MustCompile(`\d.\d.\d`)
+	if avj.buildVersion == "" || !semVerRegexp.MatchString(avj.buildVersion) {
+		return errors.ErrStartingVersionChecker.FormatError("build version is missing or of noncompliant semantic versioning")
+	}
+	return nil
+}
+
+// getJFrogVersions - obtaining the versions from JFrog website
+// **Note** this is a temporary solution to obtaining the list of versions by querying jfrog
+func (avj *agentVersionCheckJob) getJFrogVersions(name string) error {
+	b := loadPage(name)
+
+	hAnchors := htmlAnchors{}
+	err := xml.NewDecoder(bytes.NewBuffer(b)).Decode(&hAnchors)
+	if err != nil {
+		return err
+	}
+
+	avj.allVersions = hAnchors.VersionList
+	avj.latestVersion = avj.getLatestVersionFromJFrog()
+	return nil
+}
+
+// isVersionStringOlder - return true if version of str1 is older than str2
+func isVersionStringOlder(build string, latest string) bool {
+	vB := getSemVer(build)
+	vL := getSemVer(latest)
+
+	return isVersionSmaller(vB, vL)
+}
+
+// isVersionSmaller - return true if version1 smaller than version2
+func isVersionSmaller(v1 version, v2 version) bool {
+	if v1.major < v2.major {
+		return true
+	} else if v1.major == v2.major && v1.minor < v2.minor {
+		return true
+	} else if v1.major == v2.major && v1.minor == v2.minor && v1.patch < v2.patch {
```

Comment:
You could combine this a bit so you don't check major-major 2x:

if v1.major < v2.major {
		return true
} else if v1.major == v2.major {
  if v1.minor < v2.minor {
	return true
  } else if v1.minor == v2.minor && v1.patch < v2.patch {
...

Reply:
Also, go doesn't generally like all of the if/elseif things. So it would be:

if x < y {
 return true
}
if a == b {
  if c < d {
    return true
  }
  ... etc